tests/rptest/tests/group_membership_test.py

```diff
@@ -113,3 +116,187 @@ def _check_no_duplicates():
                    10,
                    backoff_sec=2,
                    err_msg="found persistent duplicates in groups listing")
+
+
+class GroupMetricsTest(RedpandaTest):
+    def __init__(self, ctx, *args, **kwargs):
+
+        extra_rp_conf = dict(default_topic_replications=3, )
+
+        super(GroupMetricsTest, self).__init__(test_context=ctx,
+                                               num_brokers=3,
+                                               extra_rp_conf=extra_rp_conf)
+        self._ctx = ctx
+        random.seed(42)
+
+    def _wait_topic_is_ready(self, topic):
+        # Wait for the partition to have a leader (`rpk produce` errors
+        # out if it tries to write data before this)
+        def partition_ready():
+            return KafkaCat(self.redpanda).get_partition_leader(
+                topic, 0)[0] is not None
+
+        wait_until(partition_ready, timeout_sec=10, backoff_sec=0.5)
+
+    def _get_offset_from_metrics(self, group):
+        metric = self.redpanda.metrics_sample("kafka_group_offset")
+        if metric is None:
+            return None
+        metric = metric.label_filter(dict(group=group))
+        res = {}
+        for sample in metric.samples:
+            res[(sample.labels['topic'],
+                 sample.labels['partition'])] = sample.value
+        return res
+
+    @cluster(num_nodes=3)
+    def test_check_value(self):
+        rpk = RpkTool(self.redpanda)
+
+        topic = "test_topic"
+        spec = TopicSpec(name=topic, partition_count=1)
+        self.redpanda.create_topic(spec)
+        self._wait_topic_is_ready(topic)
```

Comment:
I don't think that this `wait_topic_is_ready` is necessary:

1. if you let the test harness create the topic for you, then I think it also waits a little bit for topics to become ready.
2. but regardless of (1) waiting for the topic to be ready isn't a real-world approach that we should have in the tests. if the error is because the topic isn't quite ready, then it is better to retry the produce or investigate the error a little further.